domain-server/src/DomainServer.cpp

```diff
@@ -1677,6 +1686,49 @@ void DomainServer::sendHeartbeatToMetaverse(const QString& networkAddress, const
         domainObject[PUBLIC_SOCKET_ADDRESS_KEY] = networkAddressFromSettings;
     }
 
+    QJsonObject capabilities;
+
+#if defined(WEBRTC_DATA_CHANNELS)
+    QJsonObject webRTCCapabilities;
+    bool isWebRTCEnabled = _settingsManager.valueForKeyPath(WEBRTC_ENABLED_SETTINGS_KEY).toBool();
+    bool isWebRTCWSSEnabled = _settingsManager.valueForKeyPath(WEBRTC_WSS_ENABLED_SETTINGS_KEY).toBool();
+    auto acmeDomainList = _settingsManager.valueOrDefaultValueForKeyPath("acme.certificate_domains").toList();
+    QJsonArray acmeDomains;
+    for (auto&& acmeDomain : acmeDomainList) {
+        acmeDomains.push_back(acmeDomain.toMap()["domain"].toString());
+    }
+    webRTCCapabilities["enabled"] = isWebRTCEnabled;
+    webRTCCapabilities["ssl_enabled"] = isWebRTCWSSEnabled;
+    webRTCCapabilities["ssl_domains"] = acmeDomains;
+    capabilities["webrtc"] = webRTCCapabilities;
+#endif
+
+    QJsonObject prometheusCapabilities;
+    bool isExporterEnabled = _settingsManager.valueOrDefaultValueForKeyPath(ENABLE_PROMETHEUS_EXPORTER_SETTINGS_KEY).toBool();
+    int exporterPort = _settingsManager.valueOrDefaultValueForKeyPath(PROMETHEUS_EXPORTER_PORT_SETTINGS_KEY).toInt();
+    prometheusCapabilities["enabled"] = isExporterEnabled;
+    prometheusCapabilities["port"] = exporterPort;
+    capabilities["prometheus_exporter"] = prometheusCapabilities;
+
+    QJsonObject metadataCapabilities;
+    bool isMetadataExporterEnabled = _settingsManager.valueOrDefaultValueForKeyPath(ENABLE_METADATA_EXPORTER_SETTINGS_KEY).toBool();
+    int metadataExporterPort = _settingsManager.valueOrDefaultValueForKeyPath(METADATA_EXPORTER_PORT_SETTINGS_KEY).toInt();
+    metadataCapabilities["enabled"] = isMetadataExporterEnabled;
+    metadataCapabilities["port"] = metadataExporterPort;
+    capabilities["metadata_exporter"] = metadataCapabilities;
+
+    QJsonObject audioCapabilites;
+    auto audioCodecPreferenceList = _settingsManager.valueOrDefaultValueForKeyPath("audio_env.codec_preference_order").toString().split(",");
+    QJsonArray audioCodecPreference;
+    for (auto&& audioCodec : audioCodecPreferenceList) {
+        audioCodecPreference.push_back(audioCodec);
+    }
+    audioCapabilites["codec_preference_order"] = audioCodecPreference;
+    // TODO: add a list of available codecs, must be retrieved from PluginManager
```

Comment:
Does this TODO need to be filled in now? Or perhaps there's an issue for this?

Reply:
I'd say no, cause I'm not yet sure how to do it properly, and I didn't want to spend too much time on this. The audio mixer uses the PluginManager to load the codec plugins and negotiates the audio format based on that and the preference list from the settings. In the simplest case the domain server can also load an enumerate the same plugins, but it might be better to have the audio mixer set a new setting or otherwise communicate it to the domain server, for use cases where they might be separate (different servers/VMs/docker images).

It's all in context of this issue I guess https://github.com/vircadia/vircadia/issues/1421, but the part that's important for the upcoming release with the web client is the WebRTC capabilities.